packages/generators/admin/component/index.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { join } = require('path');
+const { flow, camelCase, upperFirst, lowerCase } = require('lodash');
+const fileExistsInPackages = require('../utils/fileExistsInPackages');
+const getPluginList = require('../utils/getPluginList');
+const packagesFolder = require('../utils/packagesFolder');
+
+const pascalCase = flow(camelCase, upperFirst);
+
+const prompts = [
+  {
+    type: 'list',
+    name: 'plugin',
+    message: 'Which plugin should be targeted?',
+    choices: getPluginList,
+  },
+  {
+    type: 'input',
+    name: 'name',
+    message: 'What should be the name of the component?',
+    validate: async (name, answers) => {
+      if (!name) {
+        return 'The name cannot be empty.';
+      }
+
+      return (await fileExistsInPackages(`${answers.plugin}/admin/src/components/${name}`))
+        ? 'This component already exists.'
+        : true;
+    },
+    filter: pascalCase,
+  },
+  {
+    type: 'confirm',
+    name: 'styled',
+    message: 'Is this a styled component?',
+  },
+  {
+    type: 'input',
+    name: 'htmlTag',
+    message: 'Which HTML tag should be used as a base?',
+    when: answers => answers.styled,
+    validate: htmlTag => (!htmlTag ? 'The HTML tag cannot be empty.' : true),
+    filter: lowerCase,
```

Comment:
I would like to know your thoughts on the `styled-components` generator @soupette @mfrachet @fdel-car. Since we'll have the DS base components, do you think we will need that here?

- My thought: IMO with the DS, we are not supposed to have to recreate basic (based on HTML tags) components in the admin.

Reply:
Agree we can remove that part

Reply:
We'll still have custom components to implement :) maybe we're not talking about these?

Reply:
So, just so I understand it completely, we remove this prompt about HTML tags but you'll still need to say which base component from the DS you want to style, thus another prompt replacing this one right?

I could replace the prompt above with:

```js
{
  type: 'list',
  message: 'Which base component from the DS should be styled?',
  choices: // List of DS base components
  ...
}
```

Would that be what you have in mind?

Reply:
You are right @mfrachet we will still need to create some custom components that is not in the design system. We can keep it like this 👍